### DISCUSSION TITLE
Cache local presets

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#30775] [Suggest an Idea]
### Tell us more.

We use local shared presets on Bitbucket Cloud.  

Atlassian's Bitbucket Cloud API rate limits are notoriously low, and as such, we've started to receive elevated HTTP 429 Too Many Requests errors when our repositories are fetching from our org/renovate-config shared preset repository.

Our setup leverages REDIS as a distributed cache, so I thought it could be useful if the shared presets were stored in the cache to help alleviate the load on constantly sourcing shared presets

**Comment by rarkins** [comment]
Unfortunately Bitbucket Cloud counts 304 responses against the rate limit so http caching does not help. You would need to accept that caching time (using Redis) would delay propagation time for any changes to common config

**Reply by setchy** [reply]
I think that might be an acceptable trade off worth making to have a functional renovate pipeline, vs one that's failing due to rate limits out of its control. 

Hopefully all this is resolved if/when Atlassian increase their quotas

**Reply by setchy** [reply]
Alternatively, we have ~10 shared presents in the same repository. 

Fetching each of these via HTTP will incur the quota hit per renovate pipeline execution.

I wonder if the local preset repository (ie: renovate-config) could alternatively be retrieved via a git clone instead. That has significantly higher rate limits.
